pkg/steps/pod.go

```diff
@@ -198,3 +182,65 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 		jobSpec:     jobSpec,
 	}
 }
+
+func (s *podStep) generatePodForStep() (*coreapi.Pod, error) {
+	pod := &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name: s.config.As,
+			Labels: trimLabels(map[string]string{
+				PersistsLabel:    "false",
+				JobLabel:         s.jobSpec.Job,
+				BuildIdLabel:     s.jobSpec.BuildId,
+				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				CreatedByCILabel: "true",
+			}),
+			Annotations: map[string]string{
+				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: s.name,
+			},
+		},
+		Spec: coreapi.PodSpec{
+			ServiceAccountName: s.config.ServiceAccountName,
+			RestartPolicy:      coreapi.RestartPolicyNever,
+			Containers: []coreapi.Container{
+				{
+					Name:                     s.name,
+					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
+				},
+			},
+		},
+	}
+
+	if s.config.SecretName != "" {
+		pod.Spec.Containers[0].VolumeMounts = secretVolumeMountFromConfig(s.config.SecretMountPath)
+		pod.Spec.Volumes = secretVolumeFromConfig(s.config.SecretName)
+	}
+	return pod, nil
+}
+
+func secretVolumeFromConfig(secretName string) []coreapi.Volume {
```

Comment:
nit: maybe the name `getVolumeFromSecret` or `getSecretVolume` ?

Reply:
(lol that I suggested the other one)